percona/controllers/perconapgrestore_controller.go

```diff
@@ -0,0 +1,194 @@
+	err := r.Client.Get(ctx, types.NamespacedName{Name: pgRestore.Spec.PGCluster, Namespace: request.Namespace}, pgCluster)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "get PostgresCluster")
+	}
+
+	switch pgRestore.Status.State {
+	case v2beta1.RestoreNew:
+		if restore := pgCluster.Spec.Backups.PGBackRest.Restore; restore != nil && *restore.Enabled {
+			log.Info("Waiting for another restore to finish")
+			return reconcile.Result{RequeueAfter: time.Second * 5}, nil
+		}
+
+		pgRestore.Status.State = v2beta1.RestoreStarting
+		if err := r.Client.Status().Update(ctx, pgRestore); err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "update PGRestore status")
+		}
+
+		if err := startRestore(ctx, r.Client, pgCluster, pgRestore); err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "start restore")
+		}
+
+		log.Info("Restore is starting")
+		return reconcile.Result{}, nil
+	case v2beta1.RestoreStarting:
+		job := &batchv1.Job{}
+		err := r.Client.Get(ctx, types.NamespacedName{Name: pgCluster.Name + "-pgbackrest-restore", Namespace: pgCluster.Namespace}, job)
+		if err != nil {
+			if k8serrors.IsNotFound(err) {
+				log.Info("Waiting for restore to start")
+				return reconcile.Result{RequeueAfter: time.Second * 5}, nil
+			}
+			return reconcile.Result{}, errors.Wrap(err, "get restore job")
+		}
+
+		pgRestore.Status.State = v2beta1.RestoreRunning
+		if err := r.Client.Status().Update(ctx, pgRestore); err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "update PGRestore status")
+		}
+
+		return reconcile.Result{}, nil
+	case v2beta1.RestoreRunning:
+		job := &batchv1.Job{}
+		err := r.Client.Get(ctx, types.NamespacedName{Name: pgCluster.Name + "-pgbackrest-restore", Namespace: pgCluster.Namespace}, job)
+		if err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "get restore job")
+		}
+
+		if jobCompleted(job) {
+			pgRestore.Status.State = v2beta1.RestoreSucceeded
+			if err := r.Client.Status().Update(ctx, pgRestore); err != nil {
+				return reconcile.Result{}, errors.Wrap(err, "update pgRestore status")
+			}
+
+			if err := disableRestore(ctx, r.Client, pgCluster); err != nil {
+				return reconcile.Result{}, errors.Wrap(err, "disable restore")
+			}
+
+			log.Info("Restore succeeded")
+			return reconcile.Result{}, nil
+		}
+
+		if jobFailed(job) {
+			pgRestore.Status.State = v2beta1.RestoreFailed
+			if err := r.Client.Status().Update(ctx, pgRestore); err != nil {
+				return reconcile.Result{}, errors.Wrap(err, "update pgRestore status")
+			}
+
+			if err := disableRestore(ctx, r.Client, pgCluster); err != nil {
+				return reconcile.Result{}, errors.Wrap(err, "disable restore")
+			}
+
+			log.Info("Restore failed")
+			return reconcile.Result{}, nil
+		}
+
+		log.Info("Waiting for restore to complete")
+		return reconcile.Result{RequeueAfter: time.Second * 5}, nil
+	case v2beta1.RestoreSucceeded, v2beta1.RestoreFailed:
+		return reconcile.Result{}, nil
```

Comment:
```suggestion
```

Reply:
we have exhaustiveness check in golangci-lint config. I added these to conform that. I converted it to default case tough.